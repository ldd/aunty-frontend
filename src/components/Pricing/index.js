import React from "react";

const labelData = {
  Community: {
    pricing: "Free",
    style: {
      titleColor: "#363636",
      actionClass: "is-outlined"
    },
    featureList: ["Unlimited Users", "1 GB Total Storage", "Slack Integration"]
  },
  Entreprise: {
    pricing: (
      <a
        href="mailto:aunty@lddstudios.com"
        style={{ color: "#ff3860", textDecoration: "underline" }}
      >
        Contact us
      </a>
    ),
    style: {
      titleColor: "#ff3860",
      actionClass: "is-outlined is-danger"
    },
    featureList: [
      "Unlimited Users",
      "5 GB Storage per User",
      "Unlimited Integrations",
      "Data Governance",
      "Activity Logs"
    ]
  }
};

const Feature = ({ description }) => (
  <div className="column is-full">
    <div className="field">
      <input
        className="is-checkradio"
        type="checkbox"
        checked="checked"
        readOnly
      />
      <label>{description}</label>
    </div>
  </div>
);

const PricingEntry = ({ label }) => {
  const { pricing, style, featureList = [] } = labelData[label];
  const { titleColor: color, actionClass } = style || {};
  return (
    <div className="box">
      <div
        className="column is-full   has-text-centered subtitle"
        style={{ color }}
      >
        {label}
      </div>
      <div className="column is-full"> {pricing}</div>
      {featureList.map(featureDescription => (
        <Feature
          description={featureDescription}
          key={`feature-${featureDescription}`}
        />
      ))}
      <div className="column is-full  has-text-centered">
        <div className={`button ${actionClass}`}>Choose</div>
      </div>
    </div>
  );
};

const Pricing = () => (
  <div className="hero-body">
    <div className="container">
      <div className="columns">
        <div className="column">
          <PricingEntry label="Community" />
        </div>
        <div className="column">
          <PricingEntry label="Entreprise" />
        </div>
      </div>
    </div>
  </div>
);

export default Pricing;
