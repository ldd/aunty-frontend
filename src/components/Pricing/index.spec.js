import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Pricing, { PricingEntry } from ".";

describe("Heading", () => {
  it("renders snapshot [empty]", () => {
    const component = renderer.create(<Pricing />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders default labels", () => {
    const wrapper = shallow(<Pricing />);
  });
  it("renders custom label", () => {
    const wrapper = shallow(<Pricing labels={["noLabel"]} />);
  });
  it("renders custom label", () => {
    const wrapper = shallow(<PricingEntry label={"noLabel"} />);
  });
  it("renders custom label", () => {
    const wrapper = shallow(<PricingEntry />);
  });
});
