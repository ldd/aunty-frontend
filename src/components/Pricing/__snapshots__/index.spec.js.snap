// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading renders snapshot [empty] 1`] = `
<div
  className="hero-body"
>
  <div
    className="container"
  >
    <div
      className="columns"
    >
      <div
        className="column"
      >
        <div
          className="box"
        >
          <div
            className="column is-full   has-text-centered subtitle"
            style={
              Object {
                "color": "#363636",
              }
            }
          >
            Community
          </div>
          <div
            className="column is-full"
          >
             
            Free
          </div>
          <div
            className="column is-full"
          >
            <div
              className="field"
            >
              <input
                checked="checked"
                className="is-checkradio"
                readOnly={true}
                type="checkbox"
              />
              <label>
                Unlimited Users
              </label>
            </div>
          </div>
          <div
            className="column is-full"
          >
            <div
              className="field"
            >
              <input
                checked="checked"
                className="is-checkradio"
                readOnly={true}
                type="checkbox"
              />
              <label>
                1 GB Total Storage
              </label>
            </div>
          </div>
          <div
            className="column is-full"
          >
            <div
              className="field"
            >
              <input
                checked="checked"
                className="is-checkradio"
                readOnly={true}
                type="checkbox"
              />
              <label>
                Slack Integration
              </label>
            </div>
          </div>
          <div
            className="column is-full  has-text-centered"
          >
            <div
              className="button is-outlined"
            >
              Choose
            </div>
          </div>
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="box"
        >
          <div
            className="column is-full   has-text-centered subtitle"
            style={
              Object {
                "color": "#ff3860",
              }
            }
          >
            Entreprise
          </div>
          <div
            className="column is-full"
          >
             
            <a
              href="mailto:sales@auntyapp.io"
              style={
                Object {
                  "color": "#ff3860",
                  "textDecoration": "underline",
                }
              }
            >
              Contact us
            </a>
          </div>
          <div
            className="column is-full"
          >
            <div
              className="field"
            >
              <input
                checked="checked"
                className="is-checkradio"
                readOnly={true}
                type="checkbox"
              />
              <label>
                Unlimited Users
              </label>
            </div>
          </div>
          <div
            className="column is-full"
          >
            <div
              className="field"
            >
              <input
                checked="checked"
                className="is-checkradio"
                readOnly={true}
                type="checkbox"
              />
              <label>
                5 GB Storage per User
              </label>
            </div>
          </div>
          <div
            className="column is-full"
          >
            <div
              className="field"
            >
              <input
                checked="checked"
                className="is-checkradio"
                readOnly={true}
                type="checkbox"
              />
              <label>
                Unlimited Integrations
              </label>
            </div>
          </div>
          <div
            className="column is-full"
          >
            <div
              className="field"
            >
              <input
                checked="checked"
                className="is-checkradio"
                readOnly={true}
                type="checkbox"
              />
              <label>
                Data Governance
              </label>
            </div>
          </div>
          <div
            className="column is-full"
          >
            <div
              className="field"
            >
              <input
                checked="checked"
                className="is-checkradio"
                readOnly={true}
                type="checkbox"
              />
              <label>
                Activity Logs
              </label>
            </div>
          </div>
          <div
            className="column is-full  has-text-centered"
          >
            <div
              className="button is-outlined is-danger"
            >
              Choose
            </div>
          </div>
        </div>
      </div>
      <div
        className="column "
      >
        <img
          alt="plans"
          src="/images/undraw_review_fkgn.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
